.github/workflows/docker.yml

```
@@ -0,0 +1,75 @@
+name: docker
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-and-push:
+    name: build-and-push
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Determine tag name
+        id: determine_tag
+        shell: bash
+        run: |
+          if [[ -z $GITHUB_BASE_REF ]]; then
+            # On master/stable branches.
+            branch=${GITHUB_REF#refs/heads/}
+          else
+            # On pull request branches.
+            branch=pr-$(git describe --always --match '' --abbrev=7)
+          fi
+          branch=${branch//\//-}
+          echo "tag=${branch}" >> $GITHUB_OUTPUT
```

```suggestion
          echo "tag=${tag}" >> $GITHUB_OUTPUT
```
Consider renaming the variable to match the output value.

---

Now it is still weird, just a couple of lines higher: `tag=${branch//\//-}`.